// @atlaskit/textfield/dist/es2019/storybook/examples.js
import React from 'react';
import { fireEvent } from '@testing-library/dom';
import Texfield from '../../src';
export default {
  title: 'Examples'
};
const testId = 'text-field-test-id';

const getLastTexfield = container => {
  const textField = container.querySelectorAll(`[data-testId="${testId}"]`);
  return textField[textField.length - 1];
};

export const textField = () => {
  return /*#__PURE__*/React.createElement(Texfield, {
    placeholder: "new texfield",
    testId: testId,
    width: "large"
  });
};
const interactionTasks = [{
  name: 'Focus',
  description: 'Render texfield and make input focus',
  run: async ({
    container
  }) => {
    const texfield = getLastTexfield(container);
    fireEvent.focus(texfield);
  }
}, {
  name: 'Blur',
  description: 'Render texfield and make input blur',
  run: async ({
    container
  }) => {
    const texfield = getLastTexfield(container);
    fireEvent.blur(texfield);
  }
}, {
  name: 'OnChange',
  description: 'Render texfield and make input change',
  run: async ({
    container
  }) => {
    const texfield = getLastTexfield(container);
    fireEvent.change(texfield, {
      target: {
        value: 'foo'
      }
    });
  }
}];
textField.story = {
  name: 'texfield',
  parameters: {
    performance: {
      interactions: interactionTasks
    }
  }
};
